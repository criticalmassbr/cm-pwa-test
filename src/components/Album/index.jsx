import React from 'react';

import {
  Container,
  AlbumCover,
  PlayIcon,
  AlbumName,
  AlbumDescription,
} from './styles';

function Album ({ album }) {
  return (
    <Container>
      <AlbumCover
        alt={album.name}
        src={ album.image }
      />
      <AlbumDescription data-album-desc>
        <PlayIcon className="fa fa-play-circle" />
        <AlbumName>
          { album.name }
        </AlbumName>
      </AlbumDescription>
    </Container>
  )
}

export default Album;