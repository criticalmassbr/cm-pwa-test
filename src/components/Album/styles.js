import styled from 'styled-components';

export const Container = styled.div`
  position: relative;
  height: 100%;
  width: 100%;
  box-sizing: border-box;
  cursor: pointer;
  overflow: hidden;
  box-shadow: 0px 0px 12px -1px rgba(0, 0, 0, 0.3);

  &:hover [data-album-desc] {
    opacity: 1;
  }
`;

export const AlbumCover = styled.img`
  height: 100%;
  width: 100%;
  object-fit: cover;
`;

export const AlbumDescription = styled.div`
  position: absolute;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  top: 0px;
  left: 0px;
  height: 100%;
  width: 100%;
  background-color: rgba(0,0,0,.8);
  z-index: 10;
  opacity: 0;
  transition: all .2s ease-out;
`;

export const PlayIcon = styled.i`
  color: #fff;
  font-size: 28px;
`;

export const AlbumName = styled.h3`
  color: #fff;
  font-size: 16px;
  font-weight: 400;
  margin: 0px;
  margin-top: 24px;
`;
