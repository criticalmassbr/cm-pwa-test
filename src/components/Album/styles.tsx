import styled from 'styled-components';

export const Card = styled.div`
  display: flex;
  flex-direction: column;
`;

export const Cover = styled.img`
  width: 100%;
  height: auto;
`;

export const Title = styled.div`
  text-align: center;
  font-weight: bold;
  padding-top: 1em;
`;
