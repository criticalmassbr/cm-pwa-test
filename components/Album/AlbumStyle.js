import styled from 'styled-components';
import * as C from '~/styles/Constants';
import media from "styled-media-query";

export const CardAlbum = styled.div`
  background-color: ${C.WHITE};
  border: 1px solid ${C.DARK_GREY};
  border-radius: 6px;
  box-shadow: 0 0.5em 1em -0.125em rgba(10,10,10,.1), 0 0 0 1px rgba(10,10,10,.02);
  color: ${C.DARK_GREY};
  display: block;
  padding: 1rem;
  overflow: scroll;
  font-family: 'robotolight';

  .album-sprite {
    max-width: 100%;
    height: auto;
    display: block;
    margin: 0 auto;
  }

  ${media.lessThan("768px")`
    width: 80%;
    margin: 0 auto;
  `}
`;
