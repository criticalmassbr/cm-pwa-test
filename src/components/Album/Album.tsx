import React from "react";
import { Card, Cover, Title } from "./styles";
import { IAlbum } from "./types";

function Album({ image, name }: IAlbum) {
  return (
    <Card title={name}>
      <Cover src={image} alt="album cover" />
      <Title title="album name">{name}</Title>
    </Card>
  );
}

export default Album;
