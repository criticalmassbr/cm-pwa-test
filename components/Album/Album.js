import * as GridStyle from "~/styles/Grid";
import * as Style from './AlbumStyle';

const Album = ({albumProps}) => {
  const {
    name,
    image
  } = albumProps;

  return (
    <>
      <Style.CardAlbum>
        <GridStyle.Row>
          <GridStyle.Col
            mobile={12}
          >
            <img
              src={image}
              className="album-sprite"
            />
          </GridStyle.Col>

          <GridStyle.Col
            mobile={12}
          >
            <p className="has-text-centered">
              {name}
            </p>
          </GridStyle.Col>
        </GridStyle.Row>
      </Style.CardAlbum>
    </>
  )
};

export default Album;