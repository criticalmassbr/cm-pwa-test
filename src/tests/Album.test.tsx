import { render } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import Album from '../components/Album';

test('renders artist cover and name', () => {
  const component = <Album id="1" name="Test" image="teste.jpg" />;
  const albumComponent = renderer.create(component);
  const tree = albumComponent.toJSON();
  expect(tree).toMatchSnapshot();

  const renderResult = render(component);
  const artistImage = renderResult.getByAltText(/album cover/i);
  expect(artistImage).toBeInTheDocument();
  const artistName = renderResult.getByTitle(/album name/i);
  expect(artistName).toBeInTheDocument();
});
