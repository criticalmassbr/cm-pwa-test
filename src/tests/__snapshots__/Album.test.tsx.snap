// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders artist cover and name 1`] = `
<div
  className="sc-AxjAm iPShpD"
  title="Test"
>
  <img
    alt="album cover"
    className="sc-AxirZ jAArkG"
    src="teste.jpg"
  />
  <div
    className="sc-AxiKw iNpQfc"
    title="album name"
  >
    Test
  </div>
</div>
`;
